views/js/runner/provider/qti.js

```diff
@@ -430,20 +390,12 @@ define([
 
                     self.disableItem(context.itemUri);
 
-                    // submit the response even if empty
-                    submit(true)
-                        .then(updateStats)
-                        .then(function() {
-                            self.trigger('alert.timeout', __('Time limit reached, this part of the test has ended.'), function() {
-                                computeNext('timeout', {
-                                    scope: scope,
-                                    ref: ref
-                                });
-                            });
-                        })
-                        .catch(function(err){
-                            self.trigger('error', err);
-                        });
+                    self.trigger('alert.timeout', __('Time limit reached, this part of the test has ended.'), function() {
+                        computeNext('timeout', _.merge(getItemResults(), {
+                            scope: scope,
+                            ref: ref
+                        }));
```

Comment:
I see a potential issue here, as you display the timeout alert before requesting the server.
What happen if the test taker takes a while before acknowledging the message? Or just close the browser, and open the test on another one?
Maybe should it be better to request the server, and wait for the test taker before displaying the result of the request, should it not?